src/server.rs

```diff
@@ -100,10 +100,45 @@ fn check_content_length(headers: &HeaderMap) -> Result<(), AError> {
 async fn tunnel_handler(state: &mut State) -> Result<Response<Body>, AError> {
     let headers = HeaderMap::take_from(state);
     check_content_length(&headers)?;
-
-    let full_body = body::to_bytes(Body::take_from(state)).await?;
-    let body_content = String::from_utf8(full_body.to_vec())?;
-    let sentry_instance = parse_body(body_content)?;
+    // User IP
+    let user_ip = headers
+        .get("X-Forwarded-For")
+        .and_then(|v| v.to_str().ok())
+        .and_then(|s| s.split(',').next())
+        .and_then(|s| s.trim().parse::<std::net::IpAddr>().ok())
+        .unwrap_or("0.0.0.0".parse::<std::net::IpAddr>().unwrap())
```

Comment:
Maybe we could use the request source IP there ?

It could be useful if you were to investigate a broken chain of reverse-proxy where one of them does not propagate 'X-Forwared-For' correctly.